node/src/lib.rs

```diff
@@ -73,7 +73,16 @@ pub use utils::OS_PAGE_SIZE;
 pub const MAX_THREAD_COUNT: usize = 512;
 
 fn version_string(color: bool) -> String {
-    let mut version = format!("{}-{}", env!("CARGO_PKG_VERSION"), env!("VERGEN_SHA_SHORT"));
+    let mut version = format!(
+        "{}-{}{}",
+        env!("CARGO_PKG_VERSION"),
+        env!("VERGEN_SHA_SHORT"),
+        if cfg!(feature = "enable-legacy-hashing") {
+            "-legacy"
```

Comment:
```suggestion
            "-mainnet"
```

There are probably other places we should be doing this kind of switching too...

Reply:
How aboue `casper-mainnet`? I guess others should also want to be able to call their networks _mainnet_.